File: backend/src/main/java/com/investmentresearch/service/WatchlistService.java
package com.investmentresearch.service;

import com.investmentresearch.dto.WatchlistItemRequest;
import com.investmentresearch.dto.WatchlistItemResponse;

import java.util.List;

public interface WatchlistService {

    List<WatchlistItemResponse> getWatchlist(String email);

    WatchlistItemResponse addToWatchlist(WatchlistItemRequest request, String email);

    void removeFromWatchlist(String ticker, String email);

    boolean isWatched(String ticker, String email);
}

File: frontend/src/services/watchlistService.js
import api from './api';

export const getWatchlist = () => api.get('/watchlist');
export const addToWatchlist = (data) => api.post('/watchlist', data);
export const removeFromWatchlist = (ticker) => api.delete(`/watchlist/${ticker}`);
export const isWatched = (ticker) => api.get(`/watchlist/${ticker}/status`);
